### ISSUE TITLE
SDDM very slow to start on boot

### ISSUE THREAD

**Issue by vikashiiitdm** [sddm/sddm#1075]
On my Manjaro kde plasma 5.13.5 , SDDM is taking an inordinately long time to start after boot (somewhere around 13-14 seconds). I've tried enabling haveged service at boot to decrease the startup time, but it doesn't make any difference. What can i do?

**Comment by MyNameIsMohamad** [comment]
> I'm experiencing something similar on Gentoo (relatively up-to-date, can provide version #s of relevant packages if I can figure out precisely what they are). I can seemingly get SDDM to start by entering my username at the TTY prompt and starting to type a password, which is bizarre behavior. The Xorg log hasn't helped me out so far.
> 
> @vikashiiitdm Have you learned anything interesting about this issue over the last month?

I'm on Gentoo with profile "default/linux/amd64/17.0/desktop/plasma". I'm experiencing exactly what you're mentioning.

**Comment by jaccarmac** [comment]
Did you recently update to the 5.13.5 release candidate? I'm wondering if the regression is there given I share that version with the OP. Don't particularly want to unmask 5.14 to test but that may be worth it if you're willing to deal with the resulting merge.

**Comment by MyNameIsMohamad** [comment]
Yes, I did upgrade to the 5.13.5 candidate. At this time, I am not willing to to unmask 5.14. That will change in the future.

**Comment by jaccarmac** [comment]
Some context from the Manjaro side that I wasn't aware of but is probably relevant: https://forum.manjaro.org/t/sddm-very-slow-to-show-up/46361.

The versions are slightly different and the problem seems to have been traced to SDDM rather than Plasma, but the note about btrfs caught my attention as I am using that file system.

**Comment by MyNameIsMohamad** [comment]
@jaccarmac I'm not using btrfs. I'm using ext4 for my rootfs.

**Comment by jaccarmac** [comment]
This resolved itself for me on Gentoo today. It wasn't an SDDM update so I didn't pay enormous attention, but I suspect it was a libX11 version bump.

**Comment by fancywriter** [comment]
@jaccarmac @MyNameIsMohamad @vikashiiitdm 
I have exactly the same behaviour right now on my Gentoo laptop. Don't have a clue how to resolve it.
sddm-0.18.0
libX11-1.6.7
xorg-server-1.20.3
openrc-0.38.3 (parallel init is disabled)

Actually seeing this reproduced only after I have upgraded to SSD from HDD (moved old system there without reinstalling). Perhaps old HDD was "slow enough" for me to didn't notice such thing?

**Comment by fancywriter** [comment]
Why on Earth sddm needs entropy pool or random numbers at all? It is display manager, not a cryptographic software? Am I wrong?

**Comment by ryanmusante** [comment]
@fancywriter I've wondered same thing.

**Comment by fancywriter** [comment]
It also actually duplicates #1036

**Comment by strohel** [comment]
See https://github.com/sddm/sddm/issues/1036#issuecomment-477633990 for possible kernel config option that causes the problem.

**Comment by GlitterMonster** [comment]
2025....same issue in artix.....more than 30 seconds for sddm to appear at startup.

someone in a forum mentioned it might be an entropy issue...whatever that is.

**Comment by realblobii** [comment]
> Why on Earth sddm needs entropy pool or random numbers at all? It is display manager, not a cryptographic software? Am I wrong?

secure cryptography for session keys

**Comment by Vogtinator** [comment]
> secure cryptography for session keys

Yep, and also stuff like temporary file names (which don't need to be cryptographically secure, just random enough to avoid collisions)

**Comment by realblobii** [comment]
> 2025....same issue in artix.....more than 30 seconds for sddm to appear at startup.
> 
> someone in a forum mentioned it might be an entropy issue...whatever that is.

Either spam your keyboard on sddm startup or install this thing called an entropy generator (remember to systemctl enable whichever generator you install)
